app/middleware/verifyToken.js

```diff
@@ -0,0 +1,15 @@
+const { NotExtended } = require("http-errors");
+
+function verifyToken(request, result, nextOperation) {
+    // const userToken = "HI!";
+    // get token from cookie
+    const userToken = request.cookies.userToken;
+    // if undefined
+    if (typeof userToken != 'undefined') {
+        request.token = userToken;
+        nextOperation();
+    } else {
+        // else, token is verified, & can enter the site
+        result.redirect('/');
+    }
+}
```

Comment:
Make sure at the end of the file add a line: 
```
module.exports=verifyToken;
```
so we can use this function in other files.